File: MyClinic/src/app/features/models/consultation.ts
import { Patient } from "./patient";
import { RendezVous } from "./rendez-vous";

export class Consultation {
  public static fromJson(json:any):Consultation{
    return new Consultation(
      json['idConsultation'],
      json['idRdv'],
      json['idPatient'],
      json['codeConsultation'],
      json['typeConsultation'],
      json['poidsPatient'],
      json['taillePatient'],
      json['motifConsultation'],
      json['examenConsultation'],
      json['prixConsultation'],
      json['createdAt'],
      json['modifiedAt'],
      json['patient'],
      json['rendezVous']
    )

  }

  constructor(
    public idConsultation:number,
    public idRdv:number,
    public idPatient:number,
    public codeConsultation:string,
    public typeConsultation:string,
    public poidsPatient:number,
    public taillePatient:number,
    public motifConsultation:string,
    public examenConsultation:string,
    public prixConsultation:number,
    public createdAt:Date,
    public modifiedAt:Date,
    public  patient:Patient,
    public  rendezVous:RendezVous
  ){}
}

File: gestionPatient/src/main/java/com/gestionpatient/entities/Consultation.java
package com.gestionpatient.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.util.Date;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Builder
@Entity
@Table(name = "consultation")
@EntityListeners(AuditingEntityListener.class)
public class Consultation {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "idConsultation")
    private Long idConsultation;

    @Column(name = "idRdv")
    private Long idRdv;

    @Column(name = "idPatient")
    private Long idPatient;

    @Column(name = "codeConsultation")
    private String codeConsultation;

    @Column(name = "typeConsultation")
    private String typeConsultation;

    @Column(name="poidsPatient")
    private Long poidsPatient;

    @Column(name="taillePatient")
    private Long taillePatient;

    @Column(name = "motifConsultation")
    private String motifConsultation;

    @Column(name = "examenConsultation")
    private String examenConsultation;

    @Column(name = "prixConsultation")
    private Long prixConsultation;

    @CreatedDate
    @Column(name = "createdAt")
    private Date createdAt;

    @LastModifiedDate
    @Column(name = "modifiedAt")
    private Date modifiedAt;

    @ManyToOne
    @JoinColumn(name="idPatient", referencedColumnName = "idPatient", nullable=false, insertable = false, updatable = false)
    private Patient patient;

    @OneToOne
    @JoinColumn(name="idRdv", referencedColumnName = "idRdv", nullable=false, insertable = false, updatable = false)
    private RendezVous rendezVous;

}
